client/my-sites/upgrades/domain-management/domain-connect/domain-connect-authorize-description.jsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { translate } from 'i18n-calypso';
+
+class DomainConnectAuthorizeDescription extends Component {
+	placeholder = () => {
+		return (
+			<div className="domain-connect__is-placeholder">
+				<span></span>
+				<span></span>
+				<span></span>
+			</div>
+		);
+	}
+
+	getDescription = () => {
+		const { providerId } = this.props;
+
+		switch ( providerId ) {
+			case 'g-suite':
+				return translate( 'Howdy! It looks like you want to make your domain work with the ' +
+					'{{strong}}Google G Suite email service{{/strong}}.', {
+						components: {
+							strong: <strong />
+						}
+					} );
+
+			case 'microsoft-office365':
+				return translate( 'Howdy! It looks like you want to make your domain work with the ' +
+					'{{strong}}Microsoft Office 365 service{{/strong}}.', {
+						components: {
+							strong: <strong />
+						}
+					} );
+
+			case 'zoho-mail':
+				return translate( 'Howdy! It looks like you want to make your domain work with the ' +
+					'{{strong}}Zoho Mail service{{/strong}}.', {
+						components: {
+							strong: <strong />
+						}
+					} );
+
+			default:
+				return null;
+		}
+	}
+
+	render() {
+		const { isPlaceholder } = this.props;
+
+		if ( isPlaceholder ) {
+			return this.placeholder();
+		}
+
+		const description = this.getDescription();
+		if ( description ) {
+			return (
+				<div>
```

Comment:
Why so many nested elements? You know how that impacts the rendering performance!?!?!?!!!!1111 😜 Seriously, though, a `<p>` tag should be enough?